spec/serializer/chat_message_serializer_spec.rb

```diff
@@ -57,4 +57,86 @@
       end
     end
   end
+
+  describe "#available_flags" do
+    it "returns an empty list if the user already flagged the message" do
+      reviewable = Fabricate(:reviewable_chat_message, target: message_1)
+
+      serialized =
+        described_class.new(
+          message_1,
+          scope: guardian,
+          root: nil,
+          reviewable_ids: {
+            message_1.id => reviewable.id,
+          },
+        ).as_json
+
+      expect(serialized[:available_flags]).to be_empty
+    end
+
+    it "doesn't include notify_user for self-flags" do
+      a_guardian = Guardian.new(message_1.user)
```

Comment:
```suggestion
      guardian_1 = Guardian.new(message_1.user)
```

minor nit, just for consistency, this variable name stood out to me but I don't think it needed to